activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryProcessInstancesEntityIT.java

```diff
@@ -484,6 +485,61 @@ public void shouldGetProcessInstancesFilteredByStartDate() {
         });
     }
 
+    @Test
+    public void shouldGetProcessInstancesFilteredByCompletedDate() {
+        //given
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSXXX");
+        sdf.setTimeZone(TimeZone.getTimeZone("GMT"));
+
+        Date completedDate = new Date();
+        Calendar cal = Calendar.getInstance();
+        cal.set(Calendar.SECOND, 0);
+        cal.set(Calendar.MILLISECOND, 0);
+        Date now = cal.getTime();
+
+        //set completed date as current date
+        completedDate.setTime(now.getTime());
+
+        ProcessInstance processInstanceCompletedToday = processInstanceBuilder
```

Comment:
In order to make sure that your filter is actually filtering things, it's better to include processes that should not be returned by the filter. For instance, you can have two extra processes: one completed before the desired period and another one completed after the desired period.